src/java.base/share/classes/sun/security/ssl/SSLSocketImpl.java

```diff
@@ -808,16 +808,6 @@ private void shutdownInput(
             SSLLogger.fine("close inbound of SSLSocket");
         }
 
-        // Is it ready to close inbound?
-        //
-        // No need to throw exception if the initial handshake is not started.
-        if (checkCloseNotify && !conContext.isInputCloseNotified &&
-            (conContext.isNegotiated || conContext.handshakeContext != null)) {
-
-            throw conContext.fatal(Alert.INTERNAL_ERROR,
-                    "closing inbound before receiving peer's close_notify");
-        }
-
```

Comment:
I am a little bit hesitate to close the inbound silently, especially when it is a layered socket and the write side does not stop write yet.  The shutdownInput() is used to Unsupported UnsupportedOperationException, and allowed to throw IOException.  I think application may be able to handle it properly already.

Instead, maybe we could put the block (line 821-824) in a finalize block so that the input get closed before throwing this exception.